File: frontend/src/types/compatibility.ts
export type LegacyEntityTypeExample = {
  entityId: string;
  entityName: string;
  campaignId: string;
  campaignName: string;
};

export type LegacyEntityTypeIssue = {
  legacyType: string;
  count: number;
  exampleEntities: LegacyEntityTypeExample[];
};

export type EntityTypeCompatibilityReport = {
  hasIssues: boolean;
  issueCount: number;
  issues: LegacyEntityTypeIssue[];
};

export type EntityTypeMigrationMapping = {
  legacyType: string;
  targetType: string;
};

export type EntityTypeMigrationResultItem = {
  legacyType: string;
  targetType: string;
  updatedCount: number;
};

export type EntityTypeMigrationResult = {
  updatedCount: number;
  updatedTypes: EntityTypeMigrationResultItem[];
};

File: backend/app/schemas/compatibility.py
from __future__ import annotations

from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.enums import EntityType


class LegacyEntityTypeExample(BaseModel):
    model_config = ConfigDict(extra="forbid")

    entity_id: UUID
    entity_name: str
    campaign_id: UUID
    campaign_name: str


class LegacyEntityTypeIssue(BaseModel):
    model_config = ConfigDict(extra="forbid")

    legacy_type: str
    count: int
    example_entities: list[LegacyEntityTypeExample]


class EntityTypeCompatibilityReport(BaseModel):
    model_config = ConfigDict(extra="forbid")

    has_issues: bool
    issue_count: int
    issues: list[LegacyEntityTypeIssue]


class EntityTypeMigrationMapping(BaseModel):
    model_config = ConfigDict(extra="forbid")

    legacy_type: str
    target_type: EntityType


class EntityTypeMigrationRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    mappings: list[EntityTypeMigrationMapping]


class EntityTypeMigrationResultItem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    legacy_type: str
    target_type: str
    updated_count: int


class EntityTypeMigrationResult(BaseModel):
    model_config = ConfigDict(extra="forbid")

    updated_count: int
    updated_types: list[EntityTypeMigrationResultItem]
